src/test/java/se/sundsvall/messaging/api/model/response/UserMessageTest.java

```diff
@@ -0,0 +1,107 @@
+package se.sundsvall.messaging.api.model.response;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import se.sundsvall.messaging.test.annotation.UnitTest;
+
+@UnitTest
+class UserMessageTest {
+
+	private static final String MESSAGE_ID = "messageId";
+	private static final String ISSUER = "issuer";
+	private static final String ORIGIN = "origin";
+	private static final LocalDateTime SENT = LocalDateTime.now();
+	private static final List<UserMessage.Recipient> RECIPIENTS = List.of(UserMessage.Recipient.builder().build());
+	private static final List<UserMessage.MessageAttachment> ATTACHMENTS = List.of(UserMessage.MessageAttachment.builder().build());
+	private static final String PERSON_ID = "personId";
+	private static final String MESSAGE_TYPE = "messageType";
+	private static final String CONTENT_TYPE = "contentType";
+	private static final String FILE_NAME = "fileName";
+
+	@Test
+	void userMessageConstructor() {
+		var bean = new UserMessage(MESSAGE_ID, ISSUER, ORIGIN, SENT, RECIPIENTS, ATTACHMENTS);
+
+		assertThat(bean).isNotNull().hasNoNullFieldsOrProperties();
+		assertThat(bean.messageId()).isEqualTo(MESSAGE_ID);
+		assertThat(bean.issuer()).isEqualTo(ISSUER);
+		assertThat(bean.origin()).isEqualTo(ORIGIN);
+		assertThat(bean.sent()).isEqualTo(SENT);
+		assertThat(bean.recipients()).isEqualTo(RECIPIENTS);
+		assertThat(bean.attachments()).isEqualTo(ATTACHMENTS);
+	}
+
+	@Test
+	void userMessageBuilder() {
+		var bean = UserMessage.builder()
+			.withMessageId(MESSAGE_ID)
+			.withIssuer(ISSUER)
+			.withOrigin(ORIGIN)
+			.withSent(SENT)
+			.withRecipients(RECIPIENTS)
+			.withAttachments(ATTACHMENTS)
+			.build();
+
+		assertThat(bean).isNotNull().hasNoNullFieldsOrProperties();
+		assertThat(bean.messageId()).isEqualTo(MESSAGE_ID);
+		assertThat(bean.issuer()).isEqualTo(ISSUER);
+		assertThat(bean.origin()).isEqualTo(ORIGIN);
+		assertThat(bean.sent()).isEqualTo(SENT);
+		assertThat(bean.recipients()).isEqualTo(RECIPIENTS);
+		assertThat(bean.attachments()).isEqualTo(ATTACHMENTS);
```

Comment:
I personally dislike asserting in separate methods, i think it makes the code harder to follow and I don't see much gain either.

Reply:
The gain would be to not duplicate code for one thing. Classes having many attributes to verify will lead to large test classes with many duplications. Another benefit is that there will be an error if you add an attribute, add it to the verifying method but only add it in the constructor or the builder test. Then the method that has not added the new attribute will fail. Having separate verifications and missing to add the attribute in one of them will not produce a failure in the testclass.

But in small classes it might be overkill.